File: src/main/scala/vpzomtrrfrt/creepernuggets/ItemCreeperNugget.scala
package vpzomtrrfrt.creepernuggets

import net.minecraft.creativetab.CreativeTabs
import net.minecraft.entity.EntityLivingBase
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.item.{EnumAction, Item, ItemStack}
import net.minecraft.util.{ActionResult, EnumActionResult, EnumHand}
import net.minecraft.world.World

import scala.util.Random

object ItemCreeperNugget extends Item {
  setRegistryName("creeperNugget")
  setUnlocalizedName("creeperNugget")
  setCreativeTab(CreativeTabs.FOOD)

  override def getItemUseAction(p_getItemUseAction_1_ : ItemStack): EnumAction = EnumAction.EAT

  override def getMaxItemUseDuration(p_getMaxItemUseDuration_1_ : ItemStack): Int = 32

  override def onItemUseFinish(stack: ItemStack, p_onItemUseFinish_2_ : World, p_onItemUseFinish_3_ : EntityLivingBase): ItemStack = {
    if (!p_onItemUseFinish_2_.isRemote) {
      p_onItemUseFinish_3_ match {
        case player: EntityPlayer =>
          if (!player.capabilities.isCreativeMode) {
            stack.func_190918_g(1)
          }
          val stats = player.getFoodStats
          stats.addStats(new Random().nextInt(4) + 2, 0)
          if (!stats.needFood) {
            p_onItemUseFinish_2_.createExplosion(null, player.posX, player.posY, player.posZ, 2, true)
          }
        case _ =>
      }
    }
    stack
  }

  override def onItemRightClick(world : World, player : EntityPlayer, hand : EnumHand): ActionResult[ItemStack] = {
    player.setActiveHand(hand)
    new ActionResult[ItemStack](EnumActionResult.SUCCESS, player.getHeldItem(hand))
  }
}

File: src/main/java/vpzomtrrfrt/creepernuggets/ItemCreeperNugget.java
package vpzomtrrfrt.creepernuggets;

import net.minecraft.entity.LivingEntity;
import net.minecraft.entity.player.PlayerEntity;
import net.minecraft.item.*;
import net.minecraft.sound.SoundCategory;
import net.minecraft.util.ActionResult;
import net.minecraft.util.Hand;
import net.minecraft.util.TypedActionResult;
import net.minecraft.util.UseAction;
import net.minecraft.world.World;
import net.minecraft.world.explosion.Explosion;

public class ItemCreeperNugget extends Item {
    public static ItemCreeperNugget INSTANCE = new ItemCreeperNugget();

    private ItemCreeperNugget() {
        super(new Item.Settings().group(ItemGroup.FOOD));
    }

    @Override
    public UseAction getUseAction(ItemStack stack) {
        return UseAction.EAT;
    }

    @Override
    public int getMaxUseTime(ItemStack stack) {
        return 32;
    }

    public TypedActionResult<ItemStack> use(World world, PlayerEntity user, Hand hand) {
        ItemStack itemStack = user.getStackInHand(hand);
        if (user.canConsume(false)) {
            user.setCurrentHand(hand);
            return TypedActionResult.consume(itemStack);
        } else {
            return TypedActionResult.fail(itemStack);
        }
    }

    @Override
    public ItemStack finishUsing(ItemStack stack, World world, LivingEntity user) {
        if(world.isClient) return stack;

        world.playSound(
                null,
                user.getX(),
                user.getY(),
                user.getZ(),
                user.getEatSound(stack),
                SoundCategory.NEUTRAL,
                1.0F,
                1.0F + (world.random.nextFloat() - world.random.nextFloat()) * 0.4F
        );
        if(!(user instanceof PlayerEntity) || !((PlayerEntity) user).abilities.creativeMode) {
            stack.decrement(1);
        }
        if(user instanceof PlayerEntity) {
            ((PlayerEntity) user).getHungerManager().add(world.random.nextInt(4) + 2, 0);
            if(!((PlayerEntity) user).getHungerManager().isNotFull()) {
                world.createExplosion(null, user.getX(), user.getY(), user.getZ(), 2, Explosion.DestructionType.DESTROY);
            }
        }

        return stack;
    }
}
